tpot/export_utils.py

```diff
@@ -393,6 +394,24 @@ def replace_function_calls(pipeline_list):
 
 {OUTPUT_DF}['gbc{OPERATOR_NUM}-classification'] = gbc{OPERATOR_NUM}.predict({OUTPUT_DF}.drop('class', axis=1).values)
 """.format(OUTPUT_DF=result_name, OPERATOR_NUM=operator_num, LEARNING_RATE=learning_rate, MAX_FEATURES=max_features, MIN_WEIGHT=min_weight)
+        
+        elif operator_name == '_xg_boosting':
+            
+            learning_rate = min(1., max(float(operator[3]), 0.0001))
+            subsample = min(1., max(0.1, float(operator[4])))            
+            min_child_weight = min(100, max(0, int(operator[5])))
+            max_depth = min(100, max(1, int(operator[6])))            
```

Comment:
Yeah. Haven't thought of that. We can keep just `min_child_weight` and make `max_depth` a really large  value (or `None` if this is accept by `xgboost`) to see if this works. I'll work on it.

For `min_child_weight`, (for what I know) it can't be a fraction meaning a fraction of the data. On `xgboost` the classes can have weights (created on the `DMatrix` datatype used internally by it), so it accepts fractions here just to reflect that some classes might have fractional weight.

In the case of the `sklearn` interface (`XGBClassifier`) all classes should have weight 1. So the min value here should be 1. I'll also fix this.